packages/grafana-data/src/utils/dataLinks.ts

```diff
@@ -123,3 +125,38 @@ function interpolateObjectRecursive<T extends Object>(
   }
   return obj;
 }
+
+/**
+ * Use variable map from templateSrv to determine if all variables have values
+ * @param query
+ * @param scopedVars
+ * @param getVarMap
+ */
+export function dataLinkHasAllVariablesDefined<T extends DataLink>(
+  query: T,
+  scopedVars: ScopedVars,
+  getVarMap: Function
+): { variableMap: Record<string, string | number | boolean | undefined>; allVariablesDefined: boolean } {
+  const vars = getVarMap(getStringsFromObject(query), scopedVars);
+  return {
+    variableMap: vars,
+    allVariablesDefined: Object.values(vars).every((val) => val !== undefined && val !== null),
```

Comment:
I think this is always `true` now because vars with undefined values are not added to `vars` in `getVarMap`

Reply:
Yep, this is a bug, going to fix.